test/integration/consul-container/libs/agent/container.go

```diff
@@ -0,0 +1,422 @@
+package agent
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+	"time"
+
+	dockercontainer "github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/pkg/ioutils"
+	"github.com/pkg/errors"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/consul/test/integration/consul-container/libs/utils"
+)
+
+const bootLogLine = "Consul agent running"
+const disableRYUKEnv = "TESTCONTAINERS_RYUK_DISABLED"
+
+// consulContainerNode implements the Agent interface by running a Consul agent
+// in a container.
+type consulContainerNode struct {
+	ctx            context.Context
+	client         *api.Client
+	pod            testcontainers.Container
+	container      testcontainers.Container
+	serverMode     bool
+	ip             string
+	port           int
+	datacenter     string
+	config         Config
+	podReq         testcontainers.ContainerRequest
+	consulReq      testcontainers.ContainerRequest
+	certDir        string
+	dataDir        string
+	network        string
+	id             int
+	terminateFuncs []func() error
+}
+
+func (c *consulContainerNode) GetName() string {
+	name, err := c.container.Name(c.ctx)
+	if err != nil {
+		return ""
+	}
+	return name
+}
+
+func (c *consulContainerNode) GetConfig() Config {
+	return c.config
+}
+
+func (c *consulContainerNode) GetDatacenter() string {
+	return c.datacenter
+}
+
+func (c *consulContainerNode) IsServer() bool {
+	return c.serverMode
+}
+
+func startContainer(ctx context.Context, req testcontainers.ContainerRequest) (testcontainers.Container, error) {
```

Comment:
a nitpick, maybe we can start with the exported functions and leave the non exported at the end

Reply:
yep, good call.